### PR TITLE
Test/domain test coverage

### PR DESCRIPTION
**Pull request by Ahmed-AlMancy**
# Pull Request Template

## Description
Provide a clear and concise description of what this pull request does. Include the purpose and context for the changes.
- increase test coverage for the domain module

---

## Changes Made
List the changes introduced in this pull request:

- added test cases to increase test coverage
- added test cases for edge cases

---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

### CODE REVIEW

**Review by eslam-99 — DISMISSED** [review]

File: domain/src/test/java/com/example/domain/useCase/GetUpcomingMoviesUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import org.junit.jupiter.api.assertThrows
+
+class GetUpcomingMoviesUseCaseTest {
+    private lateinit var movieRepository: MovieRepository
+    private lateinit var getUpcomingMoviesUseCase: GetUpcomingMoviesUseCase
+
+    private val upcomingMoviesData = listOf(
+        Movie(
+            1,
+            "Upcoming Action",
```
Comment (Noor-Serry) [inline domain/src/test/java/com/example/domain/useCase/GetUpcomingMoviesUseCaseTest.kt:23]: you can use helper function that will return to you movie object

File: domain/src/test/java/com/example/domain/useCase/GetUpcomingMoviesUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    @BeforeEach
+    fun setUp() {
+        movieRepository = mockk(relaxed = true)
+        getUpcomingMoviesUseCase = GetUpcomingMoviesUseCase(movieRepository)
+    }
+
+    @Test
+    fun `should call getUpcomingMovies exactly once`() = runTest {
+        coEvery { movieRepository.getUpcomingMovies() } returns emptyList()
+        getUpcomingMoviesUseCase(MovieGenre.ALL)
```
Comment (Noor-Serry) [inline domain/src/test/java/com/example/domain/useCase/GetUpcomingMoviesUseCaseTest.kt:84]: add Given , When , Then or Arrange , Act , Assert to all test function to separate each part

**Comment by Noor-Serry** [comment]
can you provide screenshot with the test coverage now

**Review by nabilBouzineDev — DISMISSED** [review]

**Review by ThoriaHamdy99 — DISMISSED** [review]

**Review by ThoriaHamdy99 — APPROVED** [review]

**Review by eslam-99 — APPROVED** [review]

File: domain/src/test/java/com/example/domain/useCase/GetAndFilterTvShowsByKeywordUseCaseTest.kt [hunk 3]
```diff
@@ -85,68 +96,209 @@ class GetAndFilterTvShowsByKeywordUseCaseTest {
...
+            )
+            val tvShow3 = TvShow(
+                id = 3,
+                name = "Kids Comedy",
+                description = "",
+                posterUrl = "",
+                productionYear = (2023).toUInt(),
+                categories = listOf(TvShowGenre.KIDS, TvShowGenre.COMEDY),
+                rating = 7.5f,
+                popularity = 9.0
```
Comment (saraashraf11) [inline domain/src/test/java/com/example/domain/useCase/GetAndFilterTvShowsByKeywordUseCaseTest.kt:226]: can you make it in separate file

Comment (Ahmed-AlMancy) [reply domain/src/test/java/com/example/domain/useCase/GetAndFilterTvShowsByKeywordUseCaseTest.kt:226]: There is a default one i didn't use as it doesn't conform to this test case

**Review by saraashraf11 — APPROVED** [review]
